### PR TITLE
[ZDF/3sat/phoenix.de] Extractor updates and fixes (#21185)

### LINKED ISSUE #21185: [3sat] New URL
```
$ youtube-dl --ignore-config --verbose https://www.3sat.de/wissen/nano/nano-21-mai-2019-102.html
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--ignore-config', u'--verbose', u'https://www.3sat.de/wissen/nano/nano-21-mai-2019-102.html']                                                                                                                   
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.05.20
[debug] Python version 2.7.16 (CPython) - Linux-5.0.0-15-generic-x86_64-with-Ubuntu-19.04-disco
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[generic] nano-21-mai-2019-102: Requesting header
WARNING: Falling back on generic information extractor.
[generic] nano-21-mai-2019-102: Downloading webpage
[generic] nano-21-mai-2019-102: Extracting information
ERROR: Unsupported URL: https://www.3sat.de/wissen/nano/nano-21-mai-2019-102.html
Traceback (most recent call last):
  File "/home/marc/bin/youtube-dl/youtube_dl/extractor/generic.py", line 2340, in _real_extract
    doc = compat_etree_fromstring(webpage.encode('utf-8'))
  File "/home/marc/bin/youtube-dl/youtube_dl/compat.py", line 2551, in compat_etree_fromstring
    doc = _XML(text, parser=etree.XMLParser(target=_TreeBuilder(element_factory=_element_factory)))
  File "/home/marc/bin/youtube-dl/youtube_dl/compat.py", line 2540, in _XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1659, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1523, in _raiseerror
    raise err
ParseError: not well-formed (invalid token): line 134, column 841
Traceback (most recent call last):
  File "/home/marc/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "/home/marc/bin/youtube-dl/youtube_dl/extractor/common.py", line 529, in extract
    ie_result = self._real_extract(url)
  File "/home/marc/bin/youtube-dl/youtube_dl/extractor/generic.py", line 3329, in _real_extract
    raise UnsupportedError(url)
UnsupportedError: Unsupported URL: https://www.3sat.de/wissen/nano/nano-21-mai-2019-102.html

```

### PR DESCRIPTION
**Pull request by FliegendeWurst**
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests: #21356 is labeled 'do-not-merge'
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Bug fix (of #21185)

---

3sat recently (?) changed their tech stack. It seems to work like ZDF's stack. In case you're wondering, they're both public TV channels in Germany. It's only logical that they would use the same tech.
Edit: turns out that phoenix.de still uses the older stack, so we have to ~keep dreisat.py for now~ move that code to phoenix.py.

Should the extractor file be renamed to zdf3sat.py or something like that?

### CODE REVIEW

File: youtube_dl/extractor/zdf.py [hunk 1]
```diff
@@ -39,10 +39,22 @@ def _extract_player(self, webpage, video_id, fatal=True):
 
 
 class ZDFIE(ZDFBaseIE):
-    _VALID_URL = r'https?://www\.zdf\.de/(?:[^/]+/)*(?P<id>[^/?]+)\.html'
+    IE_NAME = "ZDF-3sat"
```
Comment (andreygursky) [inline youtube_dl/extractor/zdf.py]: A bit confusing, since the filename is still zdf.py

File: youtube_dl/extractor/zdf.py [hunk 2]
```diff
@@ -39,10 +39,22 @@ def _extract_player(self, webpage, video_id, fatal=True):
...
+    _VALID_URL = r'https?://www\.(zdf|3sat)\.de/(?:[^/]+/)*(?P<id>[^/?]+)\.html'
```
Comment (andreygursky) [inline youtube_dl/extractor/zdf.py]: Please look at https://github.com/ytdl-org/youtube-dl/pull/21356#discussion_r299150796

Comment (FliegendeWurst) [reply youtube_dl/extractor/zdf.py]: I'll try to find a suitable test case.

Comment (LinuxOpa) [reply youtube_dl/extractor/zdf.py]: https://www.3sat.de/wissen/nano/190822-sendung-100.html for 3SAT
https://www.zdf.de/kinder/wickie-und-die-starken-maenner/der-wegezoll-102.html for ZDF
both running with the patch above
download klappt mit dem Patch

File: youtube_dl/extractor/zdf.py (hunk 1 above)
Comment (FliegendeWurst) [reply youtube_dl/extractor/zdf.py]: Should I rename the file?

**Comment by FliegendeWurst** [comment]
@dstftw: I cannot "fix" the downloader without a test case.

**Comment by LinuxOpa** [comment]
> @dstftw: I cannot "fix" the downloader without a test case.

wie kann ich helfen / how can I help ?

**Comment by FliegendeWurst** [comment]
> wie kann ich helfen / how can I help ?

If you know of any video on either site that doesn't work with the patch, feel free to post the link. I'm not certain such videos can be found by just browsing the website, since (probably) the whole tech stack was replaced.

**Comment by LinuxOpa** [comment]
> > wie kann ich helfen / how can I help ?
> 
> If you know of any video on either site that doesn't work with the patch, feel free to post the link. > 
I'm not certain such videos can be found by just browsing the website, since (probably) the whole tech stack was replaced.

**Comment by LinuxOpa** [comment]
> > wie kann ich helfen / how can I help ?
> 
> If you know of any video on either site that doesn't work with the patch, feel free to post the link.
> I'm not certain such videos can be found by just browsing the website, since (probably)
> the whole tech stack was replaced.

So far all downloads from ZDF and 3SAT work with the patch, and I download much of the above, since June I have manually inserted the patch into youtube-dl. I'm just waiting for the "3SATZDF done" to turn on the update function again. - Bis jetzt funktionieren alle Downloads von ZDF und 3SAT mit dem Patch, und ich lade viel von den Genannten herunter, seit Juni habe ich den Patch manuell eingefügt in youtube-dl. Ich warte nur noch auf das "3SATZDF erledigt" um dann die Update-Funktion wieder einzuschalten.

**Comment by Wikinaut** [comment]
I confirm that it works with (example) https://www.3sat.de/migration/3sat/robert-frank-don-t-blink-100.html .

**Comment by w4grfw** [comment]
Phoenix is now in ARD Mediathek: https://www.ardmediathek.de/phoenix/

**Comment by mojo-hakase** [comment]
So, what is the problem with this pull request and that https://github.com/ytdl-org/youtube-dl/pull/21356. Why don't they get merged. Is there anything that need to be done?

**Comment by TechupBusiness** [comment]
> So, what is the problem with this pull request and that #21356. Why don't they get merged. Is there anything that need to be done?

Yes wondering too... The issue seems to be more about decisions than actual implementation. 
Dear youtube-dl-maintainer (@dstftw ), please make your decisions so it can go on. Thanks a lot!

**Comment by pboettch** [comment]
@FliegendeWurst If I were you, I'd close this PR, rebase your changes and issue a new PR to get it integrated. Seeing the bandwidth on this repo I could understand why the maintainer might oversee updated PRs.

Thanks for your work.

**Comment by FliegendeWurst** [comment]
@dstftw I rebased this PR, can you review it again?

**Comment by rolfn** [comment]
I can also confirm that the patch works. It would be very nice to add it to the main program.

**Comment by blackjack4494** [comment]
works like a charm. added it to my fork :)

**Comment by TechupBusiness** [comment]
Yes works, would be great to see this finally merged @dstftw Thanks!

**Comment by FliegendeWurst** [comment]
Closing this PR since my repository is not available. I will open another PR.

**Comment by TechupBusiness** [comment]
> Closing this PR since my repository is not available. I will open another PR.

Thanks @FliegendeWurst !!! :clap:  We really need to get this accepted for the/a next release!

**Pull request closed without merging**